hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -688,130 +868,139 @@ public synchronized void printMutableGraph(
         infileNode.cumulativeKeysReverseTraversal =
             infileNode.totalNumberOfKeys;
       }
     }
 
     HashSet<CompactionNode> visited = new HashSet<>();
-    for (CompactionNode  infileNode : nodeList ) {
+    for (CompactionNode infileNode : nodeList) {
       if (visited.contains(infileNode)) {
         continue;
       }
       visited.add(infileNode);
-      System.out.print("traverseGraph: Visiting node " + infileNode.fileName +
-          ":\n");
+      LOG.debug("Visiting node '{}'", infileNode.fileName);
       Set<CompactionNode> nextLevel = new HashSet<>();
       nextLevel.add(infileNode);
-      Set<CompactionNode> currentLevel = new HashSet<>();
-      currentLevel.addAll(nextLevel);
+      Set<CompactionNode> currentLevel = new HashSet<>(nextLevel);
       nextLevel = new HashSet<>();
       int i = 1;
       while (currentLevel.size() != 0) {
-        LOG.warn("traverseGraph : DAG Level :" + i++);
+        LOG.debug("DAG Level {}", i++);
         for (CompactionNode current : currentLevel) {
-          LOG.warn("traverseGraph : expanding node " + current.fileName);
+          LOG.debug("Expanding node {}", current.fileName);
           Set<CompactionNode> successors =
               reverseMutableGraph.successors(current);
-          if (successors == null || successors.size() == 0) {
-            LOG.warn("traverseGraph : No successors. cumulative " +
-                "keys : " + current.cumulativeKeysReverseTraversal);
+          if (successors.size() == 0) {
+            LOG.debug("No successors. Cumulative keys: {}",
+                current.cumulativeKeysReverseTraversal);
           } else {
             for (CompactionNode oneSucc : successors) {
-              LOG.warn("traverseGraph : Adding to the next level : " +
-                  oneSucc.fileName);
-              LOG.warn("traverseGraph : " + oneSucc.fileName + "cum" + " keys"
-                  + oneSucc.cumulativeKeysReverseTraversal + "parent" + " " +
-                  current.fileName + " total " + current.totalNumberOfKeys);
+              LOG.debug("Adding to the next level: {}", oneSucc.fileName);
+              LOG.debug("'{}' cumulative keys: {}. parent '{}' total keys: {}",
+                  oneSucc.fileName, oneSucc.cumulativeKeysReverseTraversal,
+                  current.fileName, current.totalNumberOfKeys);
               oneSucc.cumulativeKeysReverseTraversal +=
                   current.cumulativeKeysReverseTraversal;
               nextLevel.add(oneSucc);
             }
           }
         }
-        currentLevel = new HashSet<>();
-        currentLevel.addAll(nextLevel);
+        currentLevel = new HashSet<>(nextLevel);
         nextLevel = new HashSet<>();
-        LOG.warn("");
       }
     }
   }
 
+  @VisibleForTesting
   public boolean debugEnabled(Integer level) {
     return DEBUG_LEVEL.contains(level);
   }
+
+  @VisibleForTesting
+  public static Logger getLog() {
+    return LOG;
+  }
```

Comment:
Why do we need these functions?

Reply:
This is for UT to set log debug level. I could simply make `LOG` public but then realized this is the way in other classes as well. Either way it works.